src/overlay.c

```diff
@@ -1,62 +1,78 @@
 // SPDX-License-Identifier: GPL-2.0-only
 #include <assert.h>
+#include <limits.h>
 #include <wlr/render/pixman.h>
 #include "labwc.h"
 #include "overlay.h"
 #include "view.h"
 
 static void
-create_overlay_rect(struct seat *seat, struct overlay_rect *rect, int fill,
-		float bg_color[4], int border_width, float border_colors[3][4])
+create_overlay_rect(struct seat *seat, struct overlay_rect *rect,
+		int bg_enabled, float bg_color[4], int border_enabled,
+		int border_width, float border_colors[3][4])
 {
 	struct server *server = seat->server;
 
-	rect->fill = fill;
-	/* For pixman renderer, always render outlines to save CPU resource */
-	if (wlr_renderer_is_pixman(server->renderer)) {
-		rect->fill = false;
+	/*
+	 * Default settings is:
+	 * pixman renderer: bg=enabled, border=disabled
+	 * Other renderers: bg=disabled, border=enabled
+	 */
+	if (bg_enabled == INT_MIN) {
+		rect->bg_enabled = !wlr_renderer_is_pixman(server->renderer);
+	} else {
+		rect->bg_enabled = bg_enabled;
 	}
+	if (border_enabled == INT_MIN) {
+		rect->border_enabled = wlr_renderer_is_pixman(server->renderer);
+	} else {
+		rect->border_enabled = border_enabled;
+	}
```

Comment:
I am not sure about this, seems to fit better into `theme.c` within `post_processing()` and supplying `server` as argument to `theme_init()`.

Reply:
How about having `struct server *server` in `struct theme`?

Reply:
I think having `server` be an argument to `theme_init()` is cleaner (and also matches better with the other stuff we call on `Reconfigure`, see `reload_config_and_theme`).

I am not sure if additionally having `server` in `struct theme` really improves things much. Note that there is also `rc.theme` globally available already (with the theme pointer being there maybe not being the best idea) so `server` would then (indirectly) be globally available as well via `rc.theme->server` which might lead to some sloppy code in the future. Thoughts @johanmalm ?

Reply:
I added `server` to `theme_init()` and `theme_builtin()` and am pretty happy with it.
I initially thought we need to add `server` to every function in `theme.c` and was hesitant about it, but it turned out only `theme_builtin()` needs `server`.

Reply:
> I am not sure if additionally having `server` in `struct theme` really improves things much. Note that there is also `rc.theme` globally available already (with the theme pointer being there maybe not being the best idea) so `server` would then (indirectly) be globally available as well via `rc.theme->server` which might lead to some sloppy code in the future. Thoughts @johanmalm ?

Agree because it invites bad design patterns. If it's proving hard (such as in callbacks, I think it's preferable to make it static variable within a translation unit (as we've done with `g_server` in files like `session-lock.c`. It avoids having a global `server`. 

For simplicity / readability, we made an exception with `rc` (and therefore also `rc.theme`) because they are set once at the start and are used so widely.

See #1744